### DISCUSSION TITLE
Endpoint custom decorator

### DISCUSSION THREAD

**Discussion by mateochr** [prkumar/uplink#291] [Q&A]
Hello, could somebody explain to me why the piece of code below returns a `functools.partial`? Thanks for the great tool.

```
import uplink


def just_a_wrapper(f):
    def wrapper(*args, **kwargs):
        print('Hello')
        return f(*args, **kwargs)

    return wrapper


class TestConsumer(uplink.Consumer):
    @just_a_wrapper
    @uplink.get('/200')
    def success(self):
        """Return success code."""


t = TestConsumer(base_url='https://mock.codes')
resp = t.success()
print(resp)
```

Hello
functools.partial(<uplink.commands.HttpMethod object at 0x7f9656f27220>, request_definition_builder_factory=<bound method RequestDefinitionBuilder._extend of <uplink.commands.RequestDefinitionBuilder object at 0x7f9656f26bf0>>)

**Comment by Ryand29** [comment]
I have the same issue, I'd like to factorize code a bit by decorating the `get`, `post` and so on method but I end up with that "issue".
